src/libraries/System.Private.CoreLib/src/System/IO/TextWriter.cs

```diff
@@ -693,7 +693,8 @@ public virtual Task WriteLineAsync(char[] buffer, int index, int count)
 
         public virtual Task WriteLineAsync()
         {
-            return WriteAsync(CoreNewLine);
+            return Task.Factory.StartNew(state => ((TextWriter) state!).WriteLine(), this,
+                CancellationToken.None, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default);
```

Comment:
Why is TextWriter being changed?  I thought the problem in the cited issue was about IndentedTextWriter.

Reply:
Yes, the issue refers to the `IndentedTextWriter`. Which only overwrittes the synchronous methods to extend the indentation function.
However, the implementation of `TextWriter.WriteLineAsync()' differs from the other `*Async' methods. Thus, the call `IndentedTextWriter.WriteLineAsync()` never executes the function `IndentedTextWriter.OutputTabs()` method, which indents the content. Because the overwritten method 'IntendedTextWriter.WriteAsync(...)' does not execute 'OutputTabs()'.

Reply:
> Which only overwrittes the synchronous methods to extend the indentation function.

Isn't that the root of the problem?  There are other things wrong with that: it means that even when the underlying text writer robustly supports async operations, any async operations issued against the IndentedTextWriter are going to end up queueing sync operations.

Seems like the right fix is to override the XxAsync methods on IndentedTextWriter appropriately.